examples/cuda/example/maxwell/simulator.cpp

```diff
@@ -384,33 +382,29 @@ struct e_field_calculator_t
   // total: 6
   //
   // read: 7; write: 2; 9 memory accesses
-  __host__ __device__ void operator()(std::size_t cell_id) const
+  __host__ __device__ void operator()(std::size_t cell_id) const __attribute__((always_inline))
   {
-    const std::size_t column = cell_id % accessor.n;
-    const bool source_owner = cell_id == source_position;
```

Comment:
We have to use `source_position` in the same space as `cell_id`. This code used both `source_position` and `cell_id` in the global index space. You are comparing `source_position` and local `cell_id` (after mapping `cell_id -= accessor.begin`). We have to map `source_position` into local index space or have the previous approach here.

Reply:
Let me know if the latest commit resolved it.

Reply:
Also feel free to re-write it as you want, important is only to try to move as many loads as early as possible.